pkg/forwarder/publisher.go

```diff
@@ -0,0 +1,38 @@
+package forwarder
+
+import (
+	"github.com/ThreeDotsLabs/watermill/message"
+
+	"github.com/pkg/errors"
+)
+
+type Publisher struct {
+	wrappedPublisher message.Publisher
+	config           Config
+}
+
+func NewPublisher(publisher message.Publisher, config Config) *Publisher {
+	return &Publisher{
+		wrappedPublisher: publisher,
+		config:           config,
+	}
+}
+
+func (p *Publisher) Publish(topic string, messages ...*message.Message) error {
+	for _, msg := range messages {
+		envelopedMsg, err := wrapMessageInEnvelope(topic, msg)
+		if err != nil {
+			return errors.Wrapf(err, "cannot wrap message, target topic: '%s', uuid: '%s'", topic, msg.UUID)
+		}
+
+		if err := p.wrappedPublisher.Publish(p.config.ForwarderTopic, envelopedMsg); err != nil {
```

Comment:
some publishers are supporting publishing multiple messages in one transaction, so probably it's better to generate all messages and them send them all